app/node/nodeHelpers.ts

```diff
@@ -118,3 +126,45 @@ export async function nodeSupportsMemo() {
     const consensusStatus = await getConsensusStatus();
     return consensusStatus.protocolVersion >= 2;
 }
+
+export interface StatusResponse {
+    status: TransactionStatus;
+    outcomes: Record<string, TransactionSummary>;
+}
+
+/**
+ * Queries the node for the status of the transaction with the provided transaction hash.
+ * The polling will continue until the transaction becomes absent or finalized.
+ * @param transactionHash the hash of the transaction to get the status for
+ * @param pollingIntervalM, optional, interval between polling in milliSeconds, defaults to every 20 seconds.
+ */
+export async function getStatus(
+    transactionHash: string,
+    pollingIntervalMs = 20000
+): Promise<StatusResponse> {
+    return new Promise((resolve) => {
+        const interval = setInterval(async () => {
+            let response;
+            try {
+                response = await getTransactionStatus(transactionHash);
+            } catch (err) {
+                // This happens if the node cannot be reached. Just wait for the next
+                // interval and try again.
+                return;
+            }
+            if (!response) {
```

Comment:
It would be nice with a comment above this line stating that this is the case if the transaction is absent.